src/attr/_make.py

```diff
@@ -680,6 +684,8 @@ def fmt_setter_with_converter(attr_name, value_var):
                                                     a.name))
         names_for_globals[val_name] = a.validator
         names_for_globals[attr_name] = a
+    if post_init:
+        lines.append("self.__post_init__()")
```

Comment:
The other way to validate dependent attributes would be for `__post_init__` to raise an exception, no?